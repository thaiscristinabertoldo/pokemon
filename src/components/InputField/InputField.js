import React, { memo } from 'react'

import { ErrorMessage } from 'formik'

import { DivStyled, InputStyled, LabelStyled, LabelRequiredStyled, ErrorMessageStyled } from './inputfield.css'

export const InputField = memo(({ field, label, required = false, ...props }) => {
  return (
    <>
      <DivStyled>
        <InputStyled type="text" {...field} {...props} />

        <LabelStyled value={field.value}>
          {label}
          {required && <LabelRequiredStyled>*</LabelRequiredStyled>}
        </LabelStyled>
      </DivStyled>
      <ErrorMessage name={field.name} component={ErrorMessageStyled} />
    </>
  )
})
