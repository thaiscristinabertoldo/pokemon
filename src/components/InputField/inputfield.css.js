import styled from 'styled-components'

import { COLORS } from '../../constants'

const DivStyled = styled.div`
  position: relative;
  border: none;
`

const InputStyled = styled.input`
  display: block;
  font-size: 16px;
  width: 100%;
  padding: 5px;
  border: none;
  border-bottom: solid 1px ${COLORS.secondary};
  color: ${COLORS.darkGrey};
  background-color: ${COLORS.background};
  &:focus {
    outline: none;
  }
`

const LabelStyled = styled.label`
  color: ${COLORS.mediumGrey};
  font-size: ${(props) => (props.value !== '' ? '14px' : '16px')};
  font-weight: normal;
  position: absolute;
  pointer-events: none;
  padding: 5px;
  top: ${(props) => (props.value !== '' ? '-20px' : '5px')};
  ${InputStyled}:focus ~ & {
    top: -20px;
    font-size: 14px;
  }
`

const ErrorMessageStyled = styled.div`
  padding: 5px;
  color: ${COLORS.red};
  font-size: 14px;
`

const LabelRequiredStyled = styled.label`
  color: ${COLORS.red};
`

export { DivStyled, InputStyled, LabelStyled, ErrorMessageStyled, LabelRequiredStyled }
