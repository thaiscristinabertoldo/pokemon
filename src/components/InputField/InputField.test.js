import React from 'react'

import { render, cleanup } from '@testing-library/react'

import { Formik, Field } from 'formik'

import { InputField } from './InputField'

afterEach(cleanup)

describe('InputField', () => {
  it('should take a snapshot', () => {
    const { asFragment } = render(
      <Formik initialValues={{}}>
        <Field name="input" label="input" component={InputField} />
      </Formik>
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it('should take a asterisk when required', () => {
    const { container } = render(
      <Formik initialValues={{}}>
        <Field name="input" label="input" required component={InputField} />
      </Formik>
    )

    expect(container).toHaveTextContent('*')
  })
})
