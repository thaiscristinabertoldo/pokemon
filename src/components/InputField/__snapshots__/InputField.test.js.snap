// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm iPiXKw"
  >
    <input
      class="sc-AxirZ dpDXnQ"
      form="[object Object]"
      name="input"
      type="text"
      value=""
    />
    <label
      class="sc-AxiKw dfvstx"
    >
      input
    </label>
  </div>
</DocumentFragment>
`;
